modules/post/windows/gather/enum_chocolatey_applications.rb

```diff
@@ -0,0 +1,102 @@
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Windows Gather Installed Application Within Chocolatey Enumeration',
+        'Description' => ' This module will enumerate all installed applications on a Windows system with chocolatey installed ',
+        'License' => MSF_LICENSE,
+        'Author' => ['Nick Cottrell <ncottrellweb[at]gmail.com>'],
+        'Platform' => ['win'],
+        'Privileged' => false,
+        'SessionTypes' => %w[meterpreter shell],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_advanced_options(
+      [
+        OptString.new('ChocoPath', [false, 'The path to the chocolatey executable if it\'s not on default path', 'choco.exe']),
+      ]
+    )
+  end
+
+  def chocopath
+    # cmd_exec('where.exe', 'choco.exe') unless chocolatey?
+
+    if chocolatey?(datastore['ChocoPath'])
+      datastore['ChocoPath']
+    elsif chocolatey?(cmd_exec('where.exe', 'choco.exe'))
+      cmd_exec('where.exe', 'choco.exe')
+    elsif chocolatey?(cmd_exec('where.exe', 'chocolatey.exe'))
+      cmd_exec('where.exe', 'chocolatey.exe')
+    else
+      # if where doesnt have any expected binaries, give false
+      false
+    end
+  end
+
+  def chocolatey?(path)
+    !!(cmd_exec(path, '-v') =~ /\d+\.\d+\.\d+/m)
+  rescue Rex::Post::Meterpreter::RequestError
+    false
+  end
+
+  def run
+    # checking that session is meterpreter and session has powershell
+    fail_with(Failure::NotFound, 'Chocolatey path not found') unless chocopath
+
+    print_status("Enumerating applications installed on #{sysinfo['Computer']}") if session.type == 'meterpreter'
+
+    # getting chocolatey version
+    choco_version = cmd_exec(chocopath, '-v')
+    print_status("Targets Chocolatey version: #{choco_version}")
+
+    # Getting results of listing chocolatey applications
+    print_status('Getting chocolatey applications.')
+
+    # checking if chocolatey is 2+ or 1.0.0
+    data = if choco_version.match(/^[10]\.\d+\.\d+$/)
+             # its version 1, use local only
+             cmd_exec(chocopath, 'list -lo')
+           elsif choco_version.match(/^(?:[2-9]|\d{2,})\.\d+\.\d+$/)
+             # its version 2 or above, no need for local
+             cmd_exec(chocopath, 'list')
+           else
+             fail_with(Failure::UnexpectedReply, "Failed to get chocolatey version. Result was unexpected: #{choco_version}")
+           end
+    print_good('Successfully grabbed all items')
+
+    # making table to better organize applications and their versions
+    table = Rex::Text::Table.new(
+      'Header' => 'Installed Chocolatey Applications',
+      'Indent' => 1,
+      'Columns' => %w[
+        Name
+        Version
+      ]
+    )
+
+    # collecting all lines that match and placing them into table.
+    items = data.scan(/(\S+)\s(\d+(?:\.\d+)*)/m)
```

Comment:
I added a fix to the regex. I didnt invlude the v into the version because I notices that only chocolatey includes the v in the version for itself. Since theres no point in including chocolatey twice, I decided to not include it. I tested it on a sample of 300 packages, and this line managed to grab all of them and ignore the rest

Reply:
Makes sense to me! Thanks!